### DISCUSSION TITLE
Symmteric boundary condition in 2D and 3D

### DISCUSSION THREAD

**Discussion by alimostafavi24** [idaholab/moose#24898] [Q&A General]
Hello everyone,

I wanted to just simulate half of the domain in 2d and quarter in 3d thanks to the symmetry of the problem.

But I do not know how to apply symmetric boundary condition on a line in 2d and on planes in 3d.

In moose I performed simulations in RZ coordinates and specifying axis of symmetry, so I'm wondering whether symmetric boundary condition is also available in moose. Simply using no flux bc is not working in my problem.

Thanks,
Ali

**Comment by GiudGiud** [comment]
Hello

This depends on the physics. For example for diffusion, symmetry is simply "no flux"
For fluid flow with finite volume, symmetry means "no flux" but also "no wall shear stress"

What physics are you solving?

Guillaume

**Reply by alimostafavi24** [reply]
I am simulating multiphase flow with coupled phase field (Cahn Hilliard) and Navier stokes. So my variables are vel_x, vel_y, vel_z, p, c (order parameter), w (chemical potential).

**Reply by GiudGiud** [reply]
All finite element right?
I would think 0 flux would be fine for those to achieve symmetry. You may have to dive into the equations to understand why it's not

**Reply by CAPRICORN12341234** [reply]
如果对称边界条件，假如对称边界垂直于x轴，则vel_x=0，\nabla p  = \nabla c = \nabla w = \nabla v_y = \nabla v_z =0
